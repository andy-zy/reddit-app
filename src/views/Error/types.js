// @flow

export type ErrorPropsT = {
  error: string,
}