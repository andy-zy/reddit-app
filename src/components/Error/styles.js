import {
  StyleSheet,
} from 'react-native'

export default StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  image: {
    height: 92.5,
    width: 128,
  },
  error: {
    marginTop: 10,
    fontSize: 16,
    color: '#EE0000',
  }
})