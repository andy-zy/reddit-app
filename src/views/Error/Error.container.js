import { connect } from 'react-redux'

import Error from './Error'
import { fetchingArticlesErrorSelector } from '../../domain/selectors'

const mapStateToProps = (state) => ({
  error: fetchingArticlesErrorSelector(state),
})

export default connect(
  mapStateToProps,
)(Error)