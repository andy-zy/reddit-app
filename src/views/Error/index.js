import Error from './Error.container'

export default Error