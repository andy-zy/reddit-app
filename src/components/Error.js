// @flow
import React, { Component } from 'react'
import {
  Text,
  View,
} from 'react-native'

import type { PropsBaseT } from '../domain/types'

import { layout, colors } from '../styles'

export default class Error extends Component<PropsBaseT> {

  static navigationOptions = {
    title: 'Error',
  }

  render() {
    const { navigation } = this.props
    const error: string = navigation.state.params.error || 'Something went wrong!'

    return (
      <View style={layout.vContainer}>
        <Text style={colors.error}>{error}</Text>
      </View>
    )
  }
}