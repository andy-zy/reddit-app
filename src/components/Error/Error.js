// @flow
import React, { Component } from 'react'
import {
  Image,
  Text,
  View,
} from 'react-native'

import type { PropsBaseT } from '../../domain/types'

import { layout } from '../../styles'
import styles from './styles'

export default class Error extends Component<PropsBaseT> {

  static navigationOptions = {
    title: 'Error',
  }

  render() {
    const { navigation } = this.props
    const error: string = navigation.state.params.error || 'Something went wrong!'

    return (
      <View style={[layout.container, styles.container]}>
        <Image
          style={styles.image}
          source={require('../../images/oops.png')}
        />
        <Text style={styles.error}>{error}</Text>
      </View>
    )
  }
}