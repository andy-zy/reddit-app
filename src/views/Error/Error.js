// @flow
import React, { Component } from 'react'
import {
  Image,
  Text,
  View,
} from 'react-native'

import type { ErrorPropsT } from './types'

import { layout } from '../../styles'
import styles from './styles'

export default class Error extends Component<ErrorPropsT> {

  static navigationOptions = {
    title: 'Error',
  }

  render() {
    const { error = 'Something went wrong!' } = this.props

    return (
      <View style={[layout.container, styles.container]}>
        <Image
          style={styles.image}
          source={require('../../images/oops.png')}
        />
        <Text style={styles.error}>{error}</Text>
      </View>
    )
  }
}