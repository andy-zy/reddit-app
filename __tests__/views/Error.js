import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import Error from '../../src/views/Error/Error'

describe('Error', () => {
  const error = 'error'

  it('renders correctly', () => {
    const tree = renderer.create(<Error error={error} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('navigationOptions title', () => {
    expect(Error.navigationOptions.title).toBe('Error')
  })
})
