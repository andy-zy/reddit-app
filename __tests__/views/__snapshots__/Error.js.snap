// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "stretch",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "padding": 10,
        "paddingBottom": 0,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/images/oops.png",
      }
    }
    style={
      Object {
        "height": 92.5,
        "width": 128,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#EE0000",
        "fontSize": 16,
        "marginTop": 10,
      }
    }
  >
    error
  </Text>
</View>
`;
